src/abstractdataframe/iteration.jl

```diff
@@ -440,15 +444,33 @@ julia> mapcols(x -> x.^2, df)
    2 │     4    144
    3 │     9    169
    4 │    16    196
+
+julia> mapcols(x -> x.^2, df, cols="y")
+4×2 DataFrame
+ Row │ x      y
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1    121
+   2 │     2    144
+   3 │     3    169
+   4 │     4    196
 ```
 """
-function mapcols(f::Union{Function, Type}, df::AbstractDataFrame)
+function mapcols(f::Union{Function, Type}, df::AbstractDataFrame; cols=All())
+    if cols == All() || cols == Colon()
+        apply = fill(true, ncol(df))
+    else
+        picked = Set(names(df, cols))
```

Comment:
Why not avoid this step by working directly with indices like `names(df, cols)` does?
```julia
    idx = index(df)[cols]
    idxs = idx isa Int ? (idx:idx) : idx
```

Reply:
Because `names(df, cols)` allows for more values of `cols` than `index(df)[cols]`. For example a function or a type.

I was considering if this is not too computationally expensive, but thought that it is OK since the most common case will be fast anyway.